### ISSUE TITLE
Videos fail to upload in staging

### ISSUE THREAD

**Issue by karina4now** [TRASH-INC/trash-inc.github.io#311]
This may be caused by the new Music Videos channel, but it's unclear. I was trying to see if using tracks I uploaded by my own account would curate to the Music Videos channel but kept receiving an Error 0 on upload. I used an uploaded track from another account and the video seemed to upload but now I can't see it in any of the channels or on my account. When I tested this with a different staging account I got the same result. I then tried using non-TFA tracks and got the same result again. 

Expected Behavior:
1. I upload a track on TFA site
2. I make a video with that track on TRASH
3. I share it
4. It shows up in NEW and in MUSIC VIDEOS

Current Behavior:
1. I upload a track on TFA site
2. I make a video with that track on TRASH
3. I share it
4. It fails to upload with an Error 0
5. If I click Try Again it still fails

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

I just used the production build of the app and created a video and successfully posted it without seeing any errors.

Which specific soundtrack (title and artist) did you choose that resulted in the failed upload?

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

Approximately how long was the video? (5 seconds? 100 seconds?)

Can you reproduce the error 0 both on WiFi and on a cell network?

Can you please post a screenshot of the error?

Did you see only "Error 0" or was there an additional error code, like "Error 621, 0" or "Error 340, 0"?

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

What version of the app are you using specifically? Try doing a two finger long press gesture and then scrolling down on the debug screen to see what the version is.

~Are you using the production version of the app or the staging version of the app?~

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

![IMG_07C26B3EF006-1](https://user-images.githubusercontent.com/12141562/81768412-dd2a6a00-948f-11ea-89af-d585740b9f87.jpeg)

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

The reported error logged in the client is simply:
```
Error Domain= Code=0 "Bad response" UserInfo={NSLocalizedDescription=Bad response}
```

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

Here's some useful stuff from API.swift `createPost(forChannels:...)` that the server send back to the client, which summed up all of this with the error code 0.

The TL;DR is that `POST https://staging.benny.trash.app/v2/posts` is failing with an HTTPS 500 error.

```
2020-05-12 20:42:15.517797-0700 TRASH[1911:309822] >>> postRequest = POST https://staging.benny.trash.app/v2/posts
2020-05-12 20:42:16.586753-0700 TRASH[1911:309822] >>> response = PMKAlamofireDataResponse(request: Optional(https://staging.benny.trash.app/v2/posts), response: Optional(<NSHTTPURLResponse: 0x2816a87a0> { URL: https://staging.benny.trash.app/v2/posts } { Status Code: 500, Headers {
    Connection =     (
        "keep-alive"
    );
    "Content-Length" =     (
        111
    );
    "Content-Type" =     (
        "application/json; charset=utf-8"
    );
    Date =     (
        "Wed, 13 May 2020 03:42:16 GMT"
    );
    Etag =     (
        "W/\"6f-qOTda2agMLHJ34ipjlusW4qSTUI\""
    );
    Server =     (
        Cowboy
    );
    Vary =     (
        "Accept-Encoding"
    );
    Via =     (
        "1.1 vegur"
    );
    "X-Powered-By" =     (
        Express
    );
} }), data: Optional(111 bytes), timeline: Timeline: { "Request Start Time": 611034135.518, "Initial Response Time": 611034136.583, "Request Completed Time": 611034136.584, "Serialization Completed Time": 611034136.584, "Latency": 1.066 secs, "Request Duration": 1.066 secs, "Serialization Durati
2020-05-12 20:42:16.588062-0700 TRASH[1911:309822] >>> response.response?.statusCode = Optional(500)
```

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

chihyuC commented in Github:

Weirdly, it doesn't seem to happen on dev server.

**Comment by karina4now** [comment]
> Arkadiusz Pęcherz commented in Jira:

I tried to reproduce it and managed to do it:

Staging build v 1.36.1 (396)iPhone XS / iOS 13.3.1WiFi network

1. pick a video from the camera roll2. pick a "sound" - in my case Banana Artist QA - 3333. Tap "Share" button

Result: Error 0 appears after a while of upload processing
[file id="ce9ac457-f1e5-4a12-9cc2-c54ea9ed4f9e"]


As I reproduced it many times, changing the style from "Recap" to any other didn't cause the error, so from that I assume that the style: Recap causes it somehow.

**Comment by karina4now** [comment]
> Arkadiusz Pęcherz commented in Jira:

I attach the log that i got from the application after the bug appeared
[file id="e70cd49d-a32c-465f-9f6a-48f262016ad1"]

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

> Weirdly, it doesn't seem to happen on dev server.

That is strange.

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

I just used the production build of the app and created a video and successfully posted it without seeing any errors.

Which specific soundtrack (title and artist) did you choose that resulted in the failed upload?

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

Approximately how long was the video? (5 seconds? 100 seconds?)

Can you reproduce the error 0 both on WiFi and on a cell network?

Can you please post a screenshot of the error?

Did you see only "Error 0" or was there an additional error code, like "Error 621, 0" or "Error 340, 0"?

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

What version of the app are you using specifically? Try doing a two finger long press gesture and then scrolling down on the debug screen to see what the version is.

~Are you using the production version of the app or the staging version of the app?~

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

![IMG_07C26B3EF006-1](https://user-images.githubusercontent.com/12141562/81768412-dd2a6a00-948f-11ea-89af-d585740b9f87.jpeg)

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

The reported error logged in the client is simply:
```
Error Domain= Code=0 "Bad response" UserInfo={NSLocalizedDescription=Bad response}
```

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

Here's some useful stuff from API.swift `createPost(forChannels:...)` that the server send back to the client, which summed up all of this with the error code 0.

The TL;DR is that `POST https://staging.benny.trash.app/v2/posts` is failing with an HTTPS 500 error.

```
2020-05-12 20:42:15.517797-0700 TRASH[1911:309822] >>> postRequest = POST https://staging.benny.trash.app/v2/posts
2020-05-12 20:42:16.586753-0700 TRASH[1911:309822] >>> response = PMKAlamofireDataResponse(request: Optional(https://staging.benny.trash.app/v2/posts), response: Optional(<NSHTTPURLResponse: 0x2816a87a0> { URL: https://staging.benny.trash.app/v2/posts } { Status Code: 500, Headers {
    Connection =     (
        "keep-alive"
    );
    "Content-Length" =     (
        111
    );
    "Content-Type" =     (
        "application/json; charset=utf-8"
    );
    Date =     (
        "Wed, 13 May 2020 03:42:16 GMT"
    );
    Etag =     (
        "W/\"6f-qOTda2agMLHJ34ipjlusW4qSTUI\""
    );
    Server =     (
        Cowboy
    );
    Vary =     (
        "Accept-Encoding"
    );
    Via =     (
        "1.1 vegur"
    );
    "X-Powered-By" =     (
        Express
    );
} }), data: Optional(111 bytes), timeline: Timeline: { "Request Start Time": 611034135.518, "Initial Response Time": 611034136.583, "Request Completed Time": 611034136.584, "Serialization Completed Time": 611034136.584, "Latency": 1.066 secs, "Request Duration": 1.066 secs, "Serialization Durati
2020-05-12 20:42:16.588062-0700 TRASH[1911:309822] >>> response.response?.statusCode = Optional(500)
```

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

chihyuC commented in Github:

Weirdly, it doesn't seem to happen on dev server.

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Arkadiusz Pęcherz commented in Jira:

I tried to reproduce it and managed to do it:

Staging build v 1.36.1 (396)iPhone XS / iOS 13.3.1WiFi network

1. pick a video from the camera roll2. pick a "sound" - in my case Banana Artist QA - 3333. Tap "Share" button

Result: Error 0 appears after a while of upload processing
[file id="ce9ac457-f1e5-4a12-9cc2-c54ea9ed4f9e"]


As I reproduced it many times, changing the style from "Recap" to any other didn't cause the error, so from that I assume that the style: Recap causes it somehow.

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Arkadiusz Pęcherz commented in Jira:

I attach the log that i got from the application after the bug appeared
[file id="e70cd49d-a32c-465f-9f6a-48f262016ad1"]

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

> Weirdly, it doesn't seem to happen on dev server.

That is strange.

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

I just used the production build of the app and created a video and successfully posted it without seeing any errors.

Which specific soundtrack (title and artist) did you choose that resulted in the failed upload?

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

Approximately how long was the video? (5 seconds? 100 seconds?)

Can you reproduce the error 0 both on WiFi and on a cell network?

Can you please post a screenshot of the error?

Did you see only "Error 0" or was there an additional error code, like "Error 621, 0" or "Error 340, 0"?

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

What version of the app are you using specifically? Try doing a two finger long press gesture and then scrolling down on the debug screen to see what the version is.

~Are you using the production version of the app or the staging version of the app?~

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

![IMG_07C26B3EF006-1](https://user-images.githubusercontent.com/12141562/81768412-dd2a6a00-948f-11ea-89af-d585740b9f87.jpeg)

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

The reported error logged in the client is simply:
```
Error Domain= Code=0 "Bad response" UserInfo={NSLocalizedDescription=Bad response}
```

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

Here's some useful stuff from API.swift `createPost(forChannels:...)` that the server send back to the client, which summed up all of this with the error code 0.

The TL;DR is that `POST https://staging.benny.trash.app/v2/posts` is failing with an HTTPS 500 error and the server is returning the message 
```
"column \"deleted_at\" of relation \"channel_post\" does not exist";
```

Client logs:
```
2020-05-12 20:49:20.965102-0700 TRASH[1918:313327] >>> postRequest = POST https://staging.benny.trash.app/v2/posts
2020-05-12 20:49:21.636522-0700 TRASH[1918:313327] >>> response = PMKAlamofireDataResponse(request: Optional(https://staging.benny.trash.app/v2/posts), response: Optional(<NSHTTPURLResponse: 0x283a3c580> { URL: https://staging.benny.trash.app/v2/posts } { Status Code: 500, Headers {
    Connection =     (
        "keep-alive"
    );
    "Content-Length" =     (
        111
    );
    "Content-Type" =     (
        "application/json; charset=utf-8"
    );
    Date =     (
        "Wed, 13 May 2020 03:49:21 GMT"
    );
    Etag =     (
        "W/\"6f-qOTda2agMLHJ34ipjlusW4qSTUI\""
    );
    Server =     (
        Cowboy
    );
    Vary =     (
        "Accept-Encoding"
    );
    Via =     (
        "1.1 vegur"
    );
    "X-Powered-By" =     (
        Express
    );
} }), data: Optional(111 bytes), timeline: Timeline: { "Request Start Time": 611034560.965, "Initial Response Time": 611034561.633, "Request Completed Time": 611034561.634, "Serialization Completed Time": 611034561.634, "Latency": 0.668 secs, "Request Duration": 0.669 secs, "Serialization Durati
2020-05-12 20:49:21.638569-0700 TRASH[1918:313327] >>> json = {
    data = "<null>";
    error =     {
        code = "-1";
        message = "column \"deleted_at\" of relation \"channel_post\" does not exist";
    };
}
```

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

chihyuC commented in Github:

Weirdly, it doesn't seem to happen on dev server.

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

drewolbrich commented in Github:

> Weirdly, it doesn't seem to happen on dev server.

That is strange.

**Comment by karina4now** [comment]
> Karina Bernacki commented in Jira:

karina4now commented in Github:

> Arkadiusz Pęcherz commented in Jira:

I tried to reproduce it and managed to do it:

Staging build v 1.36.1 (396)iPhone XS / iOS 13.3.1WiFi network

1. pick a video from the camera roll2. pick a "sound" - in my case Banana Artist QA - 3333. Tap "Share" button

Result: Error 0 appears after a while of upload processing
[file id="ce9ac457-f1e5-4a12-9cc2-c54ea9ed4f9e"]


As I reproduced it many times, changing the style from "Recap" to any other didn't cause the error, so from that I assume that the style: Recap causes it somehow.
